common/profile.go

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package common
+
+import (
+	"log"
+	"os"
+	"os/signal"
+	"runtime"
+	"runtime/pprof"
+	"syscall"
+)
+
+// Profile start profiling loop
+func Profile() {
+	cpu := make(chan os.Signal, 1)
+	signal.Notify(cpu, syscall.SIGUSR1)
+
+	memory := make(chan os.Signal, 1)
+	signal.Notify(memory, syscall.SIGUSR2)
+
+	for {
+		select {
+		case <-cpu:
+			f, err := os.Create("/tmp/skydive-cpu.prof")
+			if err != nil {
+				log.Fatal("could not create CPU profile: ", err)
+			}
+			if err := pprof.StartCPUProfile(f); err != nil {
+				log.Fatal("could not start CPU profile: ", err)
+			}
+			log.Println("CPU profiling Started")
+
+			<-cpu
+
+			pprof.StopCPUProfile()
+			log.Println("CPU profiling Stopped")
+		case <-memory:
+			// Memory Profile
+			runtime.GC()
```

Comment:
this means sending a SIGUSR2 will change the behavior of the application, and the memory dump may not be representative of actual memory typically consumed; if calling `runtime.GC()` explicitly has a favorable effect on the application then it should be invoked periodically by the application regardless of SIGUSR2.

Reply:
The goal here is to see real memory allocation in order to detect memory leak or unexpected memory allocation. If I do not call the GC I will get a lot of noise. In the future we could import this to select what we want to check.